lib/apis/v3/felixconfig.go

```diff
@@ -224,17 +224,19 @@ type FelixConfigurationSpec struct {
 
 	// FailsafeInboundHostPorts is a comma-delimited list of UDP/TCP ports and CIDRs that Felix will allow incoming traffic to host endpoints
 	// on irrespective of the security policy. This is useful to avoid accidentally cutting off a host with incorrect configuration. Each
-	// port should be specified as tcp:<cidr>:<port-number> or udp:<cidr>:<port-number>. For back-compatibility, if the protocol is not specified, it
-	// defaults to "tcp". If a CIDR is not specified, it will default to `0.0.0.0/0`. To disable all inbound host ports, use the value none.
-	// The default value allows ssh access and DHCP.
-	// [Default: tcp:0.0.0.0/0:22, udp:0.0.0.0/0:68, tcp:0.0.0.0/0:179, tcp:0.0.0.0/0:2379, tcp:0.0.0.0/0:2380, tcp:0.0.0.0/0:6443, tcp:0.0.0.0/0:6666, tcp:0.0.0.0/0:6667]
+	// port should be specified as tcp:<cidr>:<port-number> or udp:<cidr>:<port-number>. For IPv6 networks, the CIDR must also be specified between
+	// square brackets (e.g. tcp:[<cidr>]:<port-number>). For back-compatibility, if the protocol is not specified, it defaults to "tcp". If a CIDR
+	// is not specified, it will allow traffic from all addresses. To disable all inbound host ports, use the value none. The default value allows
+	// ssh access and DHCP.
+	// [Default: tcp:22, udp:68, tcp:179, tcp:2379, tcp:2380, tcp:6443, tcp:6666, tcp:6667]
 	FailsafeInboundHostPorts *[]ProtoPort `json:"failsafeInboundHostPorts,omitempty"`
 	// FailsafeOutboundHostPorts is a comma-delimited list of UDP/TCP ports and CIDRs that Felix will allow outgoing traffic from host endpoints to
 	// irrespective of the security policy. This is useful to avoid accidentally cutting off a host with incorrect configuration. Each port
-	// should be specified as tcp:<port-number> or udp:<port-number>. For back-compatibility, if the protocol is not specified, it defaults
-	// to "tcp". If a CIDR is not specified, it will default to `0.0.0.0/0`. To disable all outbound host ports, use the value none.
-	// The default value opens etcd's standard ports to ensure that Felix does not get cut off from etcd as well as allowing DHCP and DNS.
-	// [Default: tcp:0.0.0.0/0:179, tcp:0.0.0.0/0:2379, tcp:0.0.0.0/0:2380, tcp:0.0.0.0/0:6443, tcp:0.0.0.0/0:6666, tcp:0.0.0.0/0:6667, udp:0.0.0.0/0:53, udp:0.0.0.0/0:67]
+	// should be specified as tcp:<port-number> or udp:<port-number>. For IPv6 networks, the CIDR must also be specified between square brackets
```

Comment:
For the FelixConfiguration API model, I don't think we should require the IPv6 address be within brackets, right?

Because the syntax is unambiguous and has its own field:

```
- protocol: TCP
  port: 443
  net: fe80::beef
```

Reply:
It looks like the code expects it to be without brackets here as well, so the user-facing comment shouldn't suggest that it needs them.

Reply:
Do we need validation that if brackets are specified we handle them correctly?

Reply:
The issue with colons is more about how the separation works in the env configuration case where we have to specify everything as parts separated by colons. You're right in that it is not required if we are to use YAML representation, but in the string representation we need to somehow separate the port and protocol from the address. I can clarify that the brackets are only required for the env configuration case, but I'm not sure how we would like to specify differences between YAML and env usage.

Reply:
This comment only applies to the yaml / datastore case, though.